internal/job/hook/wrapper.go

```diff
@@ -202,19 +203,33 @@ func NewWrapper(opts ...WrapperOpt) (*Wrapper, error) {
 		templateType = PosixShellTemplateType
 	}
 
-	if wrap.beforeEnvPath, err = tempfile.NewClosed(
-		tempfile.WithDir(hookWrapperDir),
+	// On systems where multiple buildkite-agents are running under different
+	// users, a shared path could be owned by a different user.
+	// Creating a new temporary directory to hold the temporary files avoids
+	// this issue and makes cleanup easier.
+	tempDir, err := os.MkdirTemp(os.TempDir(), hookWrapperDir)
+	if err != nil {
+		return nil, err
```

Comment:
Should we `fmt.Errorf` this errors from this package can be very terse?

Reply:
Done.